dashboard/src/shared/utils.ts

```diff
@@ -276,3 +286,36 @@ export function getGlobalNamespaceOrNamespace(
       return "unknown";
   }
 }
+
+export function convertGrpcAuthError(e: any): CustomError | any {
+  const msg = e?.metadata?.headersMap?.["grpc-message"].toString();
+  switch (e?.code) {
+    case grpc.Code.Unauthenticated:
+      return new UnauthorizedError(msg);
+    case grpc.Code.FailedPrecondition:
+    case grpc.Code.Internal:
+      //TODO(agamez): this code shouldn't be returned by the API, but it is
+      if (["credentials", "unauthorized"].some(p => msg?.toLowerCase()?.includes(p))) {
+        return new UnauthorizedError(msg);
+      } else {
+        return new InternalServerError(msg);
+      }
+    case grpc.Code.PermissionDenied:
+      return new ForbiddenError(msg);
+    case grpc.Code.NotFound:
+      return new NotFoundError(msg);
+    case grpc.Code.AlreadyExists:
+      return new ConflictError(msg);
+    case grpc.Code.InvalidArgument:
```

Comment:
`InvalidArgument` does not like an internal server error? Maybe an error like e.g. `UnprocessableEntity` fits better?

Reply:
It is not, of course. This is tricky, as we aren't really actually using those error objects... so I just defaulted them to "server error" with no specific criteria.
Let's create error objects grouping some of them, as we may want to have them in the future. See what you think

Reply:
Agree with creating error objects that extend `CustomError` so that we can have some groups.